### PR TITLE
Fix update version author

### PR DESCRIPTION
**Pull request by SaraBoutej**

### CODE REVIEW

File: notes-service/src/main/java/org/exoplatform/wiki/jpa/JPADataStorage.java [hunk 1]
```diff
@@ -1196,7 +1196,11 @@ public void addPageVersion(Page page) throws WikiException {
       pageVersionEntity.setVersionNumber(versionNumber);
       pageVersionEntity.setName(pageEntity.getName());
       pageVersionEntity.setTitle(pageEntity.getTitle());
-      pageVersionEntity.setAuthor(pageEntity.getAuthor());
+      if(StringUtils.isNotEmpty(userName)){
+        pageVersionEntity.setAuthor(userName);
+      }else{
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/jpa/JPADataStorage.java]:
```suggestion
      }
      else{
```

File: notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java [hunk 2]
```diff
@@ -241,33 +241,14 @@ public boolean renameNote(String noteType,
...
-  /**
-   * Gets a specific version by name of the given note
-   * 
-   * @param versionName The name of the version
-   * @param note The note
-   * @return The version of the note
-   * @throws WikiException if an error occured
-   */
-  public PageVersion getVersionOfNoteByName(String versionName, Page note) throws WikiException;
+  public List<PageHistory> getVersionsHistoryOfNote(Page note, String userName) throws WikiException;
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java:252]: update javadoc

File: notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java [hunk 3]
```diff
@@ -276,7 +257,7 @@ public boolean renameNote(String noteType,
    * @param note The note
    * @throws WikiException if an error occured
    */
-  public void createVersionOfNote(Page note) throws WikiException;
+  public void createVersionOfNote(Page note, String userName) throws WikiException;
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java:262]: update javadoc

File: notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java [hunk 4]
```diff
@@ -285,7 +266,7 @@ public boolean renameNote(String noteType,
    * @param note The note
    * @throws WikiException if an error occured
    */
-  public void restoreVersionOfNote(String versionName, Page note) throws WikiException;
+  public void restoreVersionOfNote(String versionName, Page note, String userName) throws WikiException;
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java:272]: update javadoc

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java [hunk 5]
```diff
@@ -577,54 +577,32 @@ public boolean canPublicAndRetrictNote(Page currentNote, Identity currentIdentit
...
     }
-    return pageVersion;
+    return versionsHistory;
   }
 
   @Override
-  public void createVersionOfNote(Page note) throws WikiException {
-    dataStorage.addPageVersion(note);
+  public void createVersionOfNote(Page note, String userName) throws WikiException {
+    dataStorage.addPageVersion(note,userName);
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java]:
```suggestion
    dataStorage.addPageVersion(note, userName);
```

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java [hunk 6]
```diff
@@ -882,7 +882,7 @@ public boolean canPublicAndRetrictPage(Page currentPage, String currentUser) thr
   public List<PageVersion> getVersionsOfPage(Page page) throws WikiException {
     List<PageVersion> versions = dataStorage.getVersionsOfPage(page);
     if(versions == null || versions.isEmpty()) {
-      dataStorage.addPageVersion(page);
+      dataStorage.addPageVersion(page,"");
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java:885]:
```suggestion
      dataStorage.addPageVersion(page, "");
```

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java [hunk 7]
```diff
@@ -905,7 +905,7 @@ public PageVersion getVersionOfPageByName(String versionName, Page page) throws
 
   @Override
   public void createVersionOfPage(Page page) throws WikiException {
-    dataStorage.addPageVersion(page);
+    dataStorage.addPageVersion(page,"");
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java:908]:
```suggestion
    dataStorage.addPageVersion(page, "");
```

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java [hunk 8]
```diff
@@ -280,7 +280,7 @@ public Response updateNote(@ApiParam(value = "NoteBook Type", required = true) @
           note_.setName(newNoteName);
         }
         note_ = noteService.updateNote(note_, PageUpdateType.EDIT_PAGE_CONTENT_AND_TITLE, identity);
-        noteService.createVersionOfNote(note_);
+        noteService.createVersionOfNote(note_,identity.getUserId());
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java:283]:
```suggestion
        noteService.createVersionOfNote(note_, identity.getUserId());
```

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java [hunk 9]
```diff
@@ -294,15 +294,15 @@ public Response updateNote(@ApiParam(value = "NoteBook Type", required = true) @
         }
         note_.setTitle(note.getTitle());
         note_ = noteService.updateNote(note_, PageUpdateType.EDIT_PAGE_TITLE, identity);
-        noteService.createVersionOfNote(note_);
+        noteService.createVersionOfNote(note_,identity.getUserId());
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java]:
```suggestion
        noteService.createVersionOfNote(note_, identity.getUserId());
```

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java [hunk 10]
```diff
@@ -294,15 +294,15 @@ public Response updateNote(@ApiParam(value = "NoteBook Type", required = true) @
...
         if (!"__anonim".equals(identity.getUserId())) {
           WikiPageParams noteParams = new WikiPageParams(noteBookType, noteBookOwner, newNoteName);
           // noteService.removeDraftOfPage(noteParams);
         }
       } else if (!note_.getContent().equals(note.getContent())) {
         note_.setContent(note.getContent());
         note_ = noteService.updateNote(note_, PageUpdateType.EDIT_PAGE_CONTENT, identity);
-        noteService.createVersionOfNote(note_);
+        noteService.createVersionOfNote(note_,identity.getUserId());
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java]:
```suggestion
        noteService.createVersionOfNote(note_, identity.getUserId());
```

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java [hunk 11]
```diff
@@ -355,7 +355,7 @@ public Response updateNoteById(@ApiParam(value = "Note id", required = true) @Pa
           note_.setName(newNoteName);
         }
         note_ = noteService.updateNote(note_, PageUpdateType.EDIT_PAGE_CONTENT_AND_TITLE, identity);
-        noteService.createVersionOfNote(note_);
+        noteService.createVersionOfNote(note_,identity.getUserId());
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java]:
```suggestion
        noteService.createVersionOfNote(note_, identity.getUserId());
```

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java [hunk 12]
```diff
@@ -369,15 +369,15 @@ public Response updateNoteById(@ApiParam(value = "Note id", required = true) @Pa
         }
         note_.setTitle(note.getTitle());
         note_ = noteService.updateNote(note_, PageUpdateType.EDIT_PAGE_TITLE, identity);
-        noteService.createVersionOfNote(note_);
+        noteService.createVersionOfNote(note_,identity.getUserId());
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java]:
```suggestion
        noteService.createVersionOfNote(note_, identity.getUserId());
```

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java [hunk 13]
```diff
@@ -369,15 +369,15 @@ public Response updateNoteById(@ApiParam(value = "Note id", required = true) @Pa
...
         if (!"__anonim".equals(identity.getUserId())) {
           WikiPageParams noteParams = new WikiPageParams(note_.getWikiType(), note_.getWikiOwner(), newNoteName);
           noteService.removeDraftOfNote(noteParams);
         }
       } else if (!note_.getContent().equals(note.getContent())) {
         note_.setContent(note.getContent());
         note_ = noteService.updateNote(note_, PageUpdateType.EDIT_PAGE_CONTENT, identity);
-        noteService.createVersionOfNote(note_);
+        noteService.createVersionOfNote(note_,identity.getUserId());
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java]:
```suggestion
        noteService.createVersionOfNote(note_, identity.getUserId());
```

File: notes-service/src/test/java/org/exoplatform/wiki/jpa/JPADataStorageTest.java [hunk 14]
```diff
@@ -1106,11 +1108,11 @@ public void testPageVersions() throws WikiException {
...
     page.setContent("Content Page 1");
     Page createdPage = storage.createPage(wiki, wiki.getWikiHome(), page);
-
+    org.exoplatform.social.core.identity.model.Identity identity = Mockito.mock( org.exoplatform.social.core.identity.model.Identity.class);
     // When
-    storage.addPageVersion(createdPage);
+    storage.addPageVersion(createdPage, identity.getId());
     List<PageVersion> pageVersions1 = storage.getVersionsOfPage(createdPage);
-    storage.addPageVersion(createdPage);
+    storage.addPageVersion(createdPage,identity.getId());
```
Comment (azayati) [inline notes-service/src/test/java/org/exoplatform/wiki/jpa/JPADataStorageTest.java:1115]:
```suggestion
    storage.addPageVersion(createdPage, identity.getId());
```

File: notes-webapp/src/main/webapp/skin/less/notes/notes.less [hunk 15]
```diff
@@ -274,7 +274,7 @@
       .long-path-second-item {
         max-width: 42px;
       }
-      .long-path-third-item {
+      .long-path-third-item {notes-application-header
```
Comment (azayati) [inline notes-webapp/src/main/webapp/skin/less/notes/notes.less]: is this modif wanted ?

**Review by azayati — CHANGES_REQUESTED** [review]
see review

**Review by mdiha — APPROVED** [review]
LGTM

File: notes-service/src/main/java/org/exoplatform/wiki/jpa/JPADataStorage.java (hunk 1 above)
Comment (SaraBoutej) [reply notes-service/src/main/java/org/exoplatform/wiki/jpa/JPADataStorage.java]: done

File: notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java (hunk 2 above)
Comment (SaraBoutej) [reply notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java:252]: done

File: notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java (hunk 3 above)
Comment (SaraBoutej) [reply notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java:262]: Done

File: notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java (hunk 4 above)
Comment (SaraBoutej) [reply notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java:272]: Done

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java (hunk 5 above)
Comment (SaraBoutej) [reply notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java]: Done

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java (hunk 6 above)
Comment (SaraBoutej) [reply notes-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java:885]: Done

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java (hunk 7 above)
Comment (SaraBoutej) [reply notes-service/src/main/java/org/exoplatform/wiki/service/impl/WikiServiceImpl.java:908]: Done

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java (hunk 13 above)
Comment (SaraBoutej) [reply notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java]: Done

File: notes-service/src/test/java/org/exoplatform/wiki/jpa/JPADataStorageTest.java (hunk 14 above)
Comment (SaraBoutej) [reply notes-service/src/test/java/org/exoplatform/wiki/jpa/JPADataStorageTest.java:1115]: Done

File: notes-webapp/src/main/webapp/skin/less/notes/notes.less (hunk 15 above)
Comment (SaraBoutej) [reply notes-webapp/src/main/webapp/skin/less/notes/notes.less]: Removed

**Pull request merged**